src/core/exec-invoke.c

```diff
@@ -2563,12 +2617,41 @@ static int setup_exec_directory(
                 if (params->runtime_scope != RUNTIME_SCOPE_SYSTEM)
                         continue;
 
-                /* Then, change the ownership of the whole tree, if necessary. When dynamic users are used we
+                idmapping_supported = is_idmapping_supported(pp ?: p);
+                if (idmapping_supported < 0)
+                        goto fail;
```

Comment:
plaease add debug logging here, saying that the idmap support check failed. i.e. just do:

```c
if (idmapping_supported < 0) {
        log_debug_errno(idmapping_supported, "Unable to determine if ID mapping is supported on mount '%s': %m", pp ?: p);
        goto fail; 
```